src/Google.Cloud.Metadata.V1/InstanceMetadata.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Collections.Generic;
+
+namespace Google.Cloud.Metadata.V1
+{
+    /// <summary>
+    /// Metadata describing the currently running Google Compute Engine VM instance.
+    /// </summary>
+    /// <seealso cref="MetadataClient.GetInstanceMetadata"/>
+    /// <seealso cref="MetadataClient.GetInstanceMetadataAsync"/>
+    public sealed class InstanceMetadata
+    {
+        // TODO: Should we change the empty string default value to be null?
+        /// <summary>
+        /// Gets the microarchitecture this machine type should use or an empty string if the machine type can be used on any
+        /// CPU platform.
+        /// </summary>
+        public string CpuPlatform { get; }
```

Comment:
This feels like it could be an enum instead of a string. If it should definitely be a string, it would be good to give examples.